### ISSUE TITLE
Echarts elements don't present in session recording

### ISSUE THREAD

**Issue by omertalmi5** [PostHog/posthog#2798]
## Bug description
In pages including components from echarts library (https://www.npmjs.com/package/echarts).
The component show as blank area in the session recording.

*Please describe.*  
In the geef we can see the original page with echarts component below.
The echart component show as blank area in the session recording.

![Peek 2020-12-16 19-36](https://user-images.githubusercontent.com/43521606/102385549-a644a580-3fd6-11eb-947c-12a9a0adfbe8.gif)

## Expected behavior
I expect see the content of the echarts component

## How to reproduce

1. Open recording session
2. Watch the echarts component

## Environment

self-managed posthog using docker images
docker image with tag of release-1.18.0

**Comment by omertalmi5** [comment]
@macobo 
What do you think about that?
Can you add the label session recording?

**Comment by macobo** [comment]
Thanks for creating the issue! 

We use [rrweb](https://github.com/rrweb-io/rrweb) under the hood here. rrweb indeed doesn't capture canvas (or webgl) content by default. [They seem to have added functionality for canvas recently](https://github.com/rrweb-io/rrweb/blob/6c4f10e4e7ac5e099efafb671876d24fc3616aa5/docs/recipes/canvas.md) but the replayer flags are marked as unsafe and could cause issues.

Will need to think/see what we can do - want to avoid introducing a obvious security vulnerability during recording/playback. :thinking:

**Comment by omertalmi5** [comment]
@macobo Thanks about your quick response.
What do you think about adding an option for unsecured session recording for such cases?

**Comment by macobo** [comment]
Yep, but it requires a bit of planning and testing. The tricky part is not so much recording but playback as supporting canvas seems to require allowing javascript to [run in the iframe created by rrweb](https://github.com/rrweb-io/rrweb/blob/714a4fbb7349c69e4dbb4521c03e0ab5ccbf0960/src/replay/index.ts#L388-L390) which is both a security risk and might break replaying sessions in other ways.

The posthog team is about to go on a 2 week break - perhaps you're interested in testing things out and creating a PR for this? Happy to provide any assistance needed. :)

**Comment by brainysteven** [comment]
Is there any plan for this? 
I think this is super important since other session tracking tools in the market like FullStory, Logrocket can't track Canvas element.

**Comment by benjackwhite** [comment]
@brainysteven We are mostly limited here by browser technology. It is technically doable and is supported by RRWeb (the library we use to do the recording) but playing it back requires enabling scripting in such a way that PostHog becomes vulnerable, hence why we haven't supported it so far.

**Comment by benjackwhite** [comment]
Closing as duplicate of the official issue linked above
